### ISSUE TITLE
High CPU usage

### ISSUE THREAD

**Issue by denniskai1234** [AHAAAAAAA/PokemonGo-Map#1632]
As i already said in [#1223](https://github.com/AHAAAAAAA/PokemonGo-Map/pull/1223#issuecomment-234733317) the CPU gets killed when you got multiply instaces running (in my cas now 80). Since all will show all scanned informations it just overloads the server.

Hopefully someone got an idea how to solve this issue.

**Comment by keaza** [comment]
Please tell me how the hell you many to run 80 instances!? I would love to know

**Comment by denniskai1234** [comment]
I dont get what you mean? Just run them lol?

Im using a dedicated server with 32g ram intel xeon e3 1245v3 so yeah it does have good specs.

**Comment by JoshuaGarrison27** [comment]
Are you using several accounts or the same account for each? Good lord! Im a little jelly.

**Comment by keaza** [comment]
soyoustart dedicated ? what webserver are you using?

**Comment by denniskai1234** [comment]
Each map a single account.

Nope directly from ovh im keeping the cheap old goodies there :) Got still like 4 of them.

**Comment by keaza** [comment]
are you using the built in webserver? if so try using a nginx frontend that proxys's i found it reduces the load quite a bit

if you need help with it add my skype: luke.watson52

**Comment by denniskai1234** [comment]
How you want to use nginx to proxy that? Also the database got like 40-50k pokemons in it so i dont think that will solve the issue :/

**Comment by keaza** [comment]
can you send me your script you use to start all the 80 instances ? (remove the passwords ofc)

**Comment by denniskai1234** [comment]
Just a bash script which also starts the script just on a different port and username/password combination? Nothing special.

**Comment by keaza** [comment]
could be why, your starting 80 servers with there own webservers. use the -ns at the end of them to tunr the webserver off on them will help load

**Comment by denniskai1234** [comment]
As long no one opens a website its fine. When just 1 person view the main page the process is using 200% of the cpu.

**Comment by keaza** [comment]
care to share the site?

**Comment by denniskai1234** [comment]
Im using the old old master version without any features since that uses only 1-2% of the cpu.

But here is a screenshot which i took from one city: http://i.epvpimg.com/AhMzh.jpg

**Comment by keaza** [comment]
https://imgur.com/a/sb0cJ

this is on my HOME server with a 80/20mbit connection. 1 thread each area. and 10 steps. i use -ns at the end of each instance. and look at my loads. I have 55 instances currently running

**Comment by ericpadilla** [comment]
why would someone need to run 80 instances?

**Comment by keaza** [comment]
and this is a intel celeron. FYI

**Comment by denniskai1234** [comment]
@keaza 

How much pokemons in your database?

@ericpadilla 

Im setting the maps up for other people who dont know how / just play pokemon dont even know how to use a PC ;)

**Comment by keaza** [comment]
when its finished the render run i will give you a awnser to that,

**Comment by keaza** [comment]
I just have a connection bottleneck

**Comment by denniskai1234** [comment]
@coolaj86

Im not sure it it would solve this problem? It still shows tons of pokemons and so on which overloads the webserver.

**Comment by keaza** [comment]
@denniskai1234  you are trying to run 80 webservers. by telling it a IP and a port and not the -ns flag its binding to them ports etc. and creating tasks that do not need to be run

**Comment by denniskai1234** [comment]
@keaza 

That dosent solve my issue, even when just i browse it it uses 200% of the cpu cause it loads 50k pokemons to the map. The unused websites are using like 2% of the cpu or less.

**Comment by coolaj86** [comment]
@denniskai1234 realistically there's no reason for the server to be caching that data in the first place. That should be handled by the client. I'm working on a node version with a fork of this repo's web ui and I'm storing the cache in each client individually.

**Comment by denniskai1234** [comment]
@coolaj86

Okay now i got it, then it sounds good :)

**Comment by keaza** [comment]
Yeah i just got upto 80k pokemon in my database, and the account got banned, and looks like i exceeded the API limit

**Comment by Sharke** [comment]
Lol 80 instances and then complaining about CPU usage.. I have an E5-2620 and am running 5. 80 is absurd.

**Comment by denniskai1234** [comment]
@Sharke 

With the new one its not working. Old version it works without any issue, 88% idle.

**Comment by keaza** [comment]
@denniskai1234 have you tried to use -ns? If not please try it

**Comment by Mouchee** [comment]
@keaza 
@denniskai1234 

I'm encountering the same CPU usage (and lot of lags client-side) problem when using latest updates.
Old ones are perfectly fine, latest ones however are massively lagging: same machine, and exact same configuration.
For info, I'm using -ns option for all my instances.

Maybe the addition of scanned zones is creating a crazy amount of map items, leading to high CPU usage ? (Didn't have this feature on the old version)
I can't say which build I'm using is the "old" one but it's quite recent: installed it yesterday during afternoon from developer branch.

**Comment by denniskai1234** [comment]
A quote from the linked pull request above:

> @denniskai1234 I think that's unveiling other problem, that is too many items being returned by /raw_data and shown on the map. As @jangrewe already mentioned, the /raw_data endpoint should only return the pokemons inside the viewport. That would reduce those high cost queries significantly. But I think that deserves a different PR.


Explain this issue pretty much,

**Comment by jangrewe** [comment]
Have a look at #1654, that will at least fix Pokemon and Scanned Location size.
Pokestops and Gyms still need to be worked on, and Pokestops will be the most rewarding one!

**Comment by ghormoon** [comment]
/raw_data will not be only problem as having multiple -ns instances and one -os has them all hyped up on the CPU ... actually the -os least
running 4 instances with -t 15 and each takes saround 50% cpu

**Comment by ghormoon** [comment]
using cprofile module i'm getting 
39 38.976 0.999 38.976 0.999 {time.sleep}
out of 41s usertime, 24s cputime.
even if all the rest was cputime in other methods, this leaves 22s of cputime for time.sleep, maybe it's used improperly somewhere? or am I just misinterpertting the results?

**Comment by denniskai1234** [comment]
@ghormoon With multithreading i dont even need to try it, will crash the cpu anyway.

**Comment by ghormoon** [comment]
experimenting with the profiles it seems to me that 50% of time waiting for requests is in CPU, that should be much less for blocking waits imho, but I can't find out how to improve it :( it either hangs on select.select or time.sleep but eats CPU.

**Comment by vrgomes** [comment]
Same here: 
I'm encountering the same CPU usage (and lot of lags client-side) problem when using latest updates.
Old ones are perfectly fine, latest ones however are massively lagging: same machine, and exact same configuration.
For info, I'm using -ns option for all my instances.
----

The old version is much better in performance CPU, Network Bandwidth.

**Comment by frumpymoons** [comment]
My ram gets filled constantly on Debian 8. Is this the case for you too? Older versions worked fine.

**Comment by vrgomes** [comment]
Same here.

**Comment by jangrewe** [comment]
To follow up on this, my 7 searchers (with -ns) each use ~6% CPU (AMD 8350), while the 3 frontends (with -os) each use ~60% CPU, and only because i've scaled up to 3 of them now - with only a couple of users on the map. I think the web-serving part is currently the biggest CPU hog, it's ridiculous how much it uses for serving a couple of simple queries from the DB...
